overlord/restart/restart.go

```diff
@@ -402,6 +438,34 @@ func restartParametersFromChange(chg *state.Change) (*RestartParameters, error)
 	return &rp, nil
 }
 
+func boundaryPathFromStatus(status state.Status) RestartBoundaryPath {
+	switch status {
+	case state.DoStatus, state.DoingStatus, state.DoneStatus:
+		return RestartBoundaryPathDo
+	case state.UndoStatus, state.UndoingStatus, state.UndoneStatus:
+		return RestartBoundaryPathUndo
+	default:
+		return 0
+	}
+}
+
+func taskIsRestartBoundary(t *state.Task, path RestartBoundaryPath) bool {
+	var boundaryStr string
+	if err := t.Get("restart-boundaries", &boundaryStr); err != nil {
```

Comment:
I think this should be singular restart-boundary

Reply:
Makes sense, changed